fhirstore/search/urlparser.py

```diff
@@ -0,0 +1,138 @@
+def pre_process_params(url_args):
+    search_args = url_to_dict(url_args)
+    processed_params = defaultdict(list)
+    if search_args == {}:
+        processed_params = {}
+
+    for key, value in search_args.items():
+        if len(value) == 1:
+            parsed_key, parsed_value = parse_comma(key, value[0])
+            processed_params.update({parsed_key: parsed_value})
+        else:
+            for element in value:
+                parsed_key, parsed_value = parse_comma(key, element)
+                if parsed_key == "multiple":
+                    processed_params[parsed_key] = parsed_value
+                # /                    processed_params["multiple"][sub_key].append(parsed_dict["multiple"][sub_key])
+                else:
+                    processed_params[parsed_key].append(parsed_value[0])
+    return processed_params
+
+
+class SearchArguments:
+    def __init__(self, url_args, resource_type):
+        self.args = pre_process_params(url_args)
+        self.resource_type = resource_type
+
+        # core args
+        self.core_args = {}
+        self.sort = None
+
+        # formatting args
+        self.elements = None
+        self.include = False
+
+        # meta args
+        self.summary = False
+        self.is_summary_count = False
+        self.offset = 0
+        self.result_size = 100
+
+
+class UrlParser:
+    def __init__(self):
+        self.core_args = {}
+
+    def sort_params(self, args):
+        has_sort = None
+        if "_sort" in args:
+            has_sort = args["_sort"]
+        elif "multiple" in args:
+            has_sort = args["multiple"].get("_sort", None)
+        # if there is a sorting argument, process it to handle a change of sorting order
+        if has_sort:
+            sorting_params = []
+            for argument in has_sort:
+                # find a "-" before the argument. It indicates a sorting order different from default
+                has_minus = re.search(r"^-(.*)", argument)
+                if has_minus:
+                    # if the argument is -score, sort by ascending order (i.e. ascending relevance)
+                    if has_minus.group(1) == "_score":
+                        sorting_params.append({has_minus.group(1): {"order": "asc"}})
+                    # for any other argument, sort by descending order
+                    else:
+                        sorting_params.append({has_minus.group(1): {"order": "desc"}})
+                # if there is no "-", use order defaults defined in elasticsearch
+                else:
+                    sorting_params.append(argument)
+            return sorting_params
+
+    def include_params(self, args):
+        include = None
+        if "_include" in args:
+            include = [re.search(r"(.*):(.*)", args["_include"][0]).group(2)]
+
+        elif "multiple" in args and "_include" in args["multiple"]:
+            attributes = args["multiple"].get("_include", None)
```

Comment:
You know that "_include" is in args["multiple"] so you can write args["multiple"]["_include"]